Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -968,6 +975,43 @@ static void webkitWebViewBaseUnmap(GtkWidget* widget)
     webkitWebViewBaseScheduleUpdateActivityState(webViewBase, ActivityState::IsVisible);
 }
 
+static bool shouldForwardKeyEvent(WebKitWebViewBase* webViewBase, GdkEvent* event)
+{
+#if USE(GTK4)
+    return event && webViewBase->priv->keyEventsToPropagate.removeFirst(event);
+#else
+    return std::exchange(webViewBase->priv->shouldForwardNextKeyEvent, false);
+#endif
+}
+
+static bool shouldForwardWheelEvent(WebKitWebViewBase* webViewBase, GdkEvent* event)
+{
+#if USE(GTK4)
+    if (!event || webViewBase->priv->wheelEventsToPropagate.isEmpty())
+        return false;
+    if (gdk_event_get_event_type(event) != GDK_TOUCHPAD_HOLD) {
+        // Some events that were meant for delayed propagation may have been coalesced/compressed by GDK into the current event:
+        // https://docs.gtk.org/gtk4/migrating-3to4.html#adapt-to-gdkevent-api-changes
+        // There's no way to both propagate and stop the current event, so just clear out these obsolete events from the to-be-propagated vector.
+        unsigned length = 0;
+        GUniquePtr<GdkTimeCoord> history(gdk_event_get_history(event, &length));
```

Comment:
Huh, this is the best way to write it currently, but it's very confusing and would fail if GdkTimeCoord were to ever gain a more specialized deleter in the future. That's very unlikely to happen in GTK 4, but who knows what the future holds. Problem is it looks like you have a pointer to delete a GdkTimeCoord, but what you actually have is a pointer to delete an array of unowned GdkTimeCoords.

I don't want to ask you to spend much time on this, but please do a quick/easy test to see if `GUniquePtr<GdkTimeCoord, GFreeDeleter>` would work. You would need to add `GFreeDeleter` in GUniquePtr.h. It would be exactly the same as GPtrDelete except there would be no specializations for it, so you can clearly guarantee what the deleter func is. This is optional; if you have any trouble doing so, then don't worry about it.

Reply:
Is this what you were thinking?

Reply:
Yes, perfect.